### PR TITLE
feat: add support of eth_secp256k1 and eth_bls algo for v0.47

### PR DESCRIPTION
**Pull request by Pythonberg1997**
### Description

This pr is to add support of eth_secp256k1 and eth_bls algo for v0.47.

And due to tendermint is upgraded to cometbft, we need to migrate the dependency on [evmos](https://github.com/evmos/ethermint) to this repository.

### Changes

Notable changes:
* add eth_secp256k1 and eth_bls in keys algo
* change default algo to eth_secp256k1

### CODE REVIEW

File: crypto/keys/eth/ethsecp256k1/privkey.go [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+// UnmarshalAminoJSON overrides Amino JSON marshalling.
+func (privKey *PrivKey) UnmarshalAminoJSON(bz []byte) error {
+	return privKey.UnmarshalAmino(bz)
+}
+
+// Sign creates a recoverable ECDSA signature on the secp256k1 curve over the
+// provided hash of the message. The produced signature is 65 bytes
+// where the last byte contains the recovery ID.
+func (privKey *PrivKey) Sign(digestBz []byte) ([]byte, error) {
+	if len(digestBz) != crypto.DigestLength {
```
Comment (guagualvcha) [inline crypto/keys/eth/ethsecp256k1/privkey.go:96]: image
There is a TODO in the ethermint code, can we return error here if the length is not expected, it is more safe.

File: crypto/keys/eth/ethsecp256k1/pubkey.go [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+func (pubKey *PubKey) UnmarshalAminoJSON(bz []byte) error {
+	return pubKey.UnmarshalAmino(bz)
+}
+
+// VerifySignature verifies that the ECDSA public key created a given signature over
+// the provided message. It will calculate the Keccak256 hash of the message
+// prior to verification.
+//
+// CONTRACT: The signature should be in [R || S] format.
+func (pubKey *PubKey) VerifySignature(msg, sig []byte) bool {
```
Comment (guagualvcha) [inline crypto/keys/eth/ethsecp256k1/pubkey.go:90]: `verifySignatureAsEIP712 ` is not enabled in our version, do you know why?

File: crypto/types/compact_bit_array_test.go [hunk 3]
```diff
@@ -89,8 +89,8 @@ func TestJSONMarshalUnmarshal(t *testing.T) {
 	bA6.SetIndex(15, true)
 
 	testCases := []struct {
-		bA           *CompactBitArray
-		marshalledBA string
+		bA          *CompactBitArray
+		marshaledBA string
```
Comment (guagualvcha) [inline crypto/types/compact_bit_array_test.go:93]: why change this?

File: math/dec_internal_test.go [hunk 4]
```diff
@@ -78,7 +78,7 @@ func (s *decimalInternalTestSuite) TestDecMarshalJSON() {
 		name    string
 		d       LegacyDec
 		want    string
-		wantErr bool // if wantErr = false, will also attempt unmarshaling
+		wantErr bool // if wantErr = false, will also attempt unmarshalling
```
Comment (guagualvcha) [inline math/dec_internal_test.go:81]: I think `marshaling` is a correct wording.. Not sure why you have to change it?

File: crypto/keys/eth/ethsecp256k1/pubkey.go (hunk 2 above)
Comment (Pythonberg1997) [reply crypto/keys/eth/ethsecp256k1/pubkey.go:90]: It's not suitable for us

**Review by guagualvcha — APPROVED** [review]

**Pull request merged**